### PR TITLE
Set better defaults for registering meta

### PR DESCRIPTION
**Pull request by srtfisher**
Also adds more hook assertions.

### CODE REVIEW

File: src/mantle/testing/expectation/class-expectation.php [hunk 1]
```diff
@@ -167,16 +167,14 @@ public function validate() {
...
+			if ( isset( $this->return_value_callback ) ) {
 				foreach ( $this->record_stop as $record ) {
-					PHPUnit::assertSame(
-						[ $this->return_value ],
-						$record,
+					PHPUnit::assertTrue(
+						call_user_func_array( $this->return_value_callback, $record ),
 						sprintf(
-							'Failed asserting that hook [%s] return value %s is identical to %s.',
+							'Failed asserting that hook\'s [%s] return value %s matches the expected return value.',
```
Comment (mogmarsh) [inline src/mantle/testing/expectation/class-expectation.php:175]: 'Failed asserting that hook [%s]\'s return value %s matches the expected return value.',

File: tests/testing/concerns/test-interacts-with-hooks.php [hunk 2]
```diff
@@ -47,4 +47,66 @@ public function test_hook_added_declaration() {
...
+
+		$this->assertIsArray( apply_filters( 'array_hook_to_add', 'not_array' ) );
+	}
+
+	public function test_hook_return_string() {
+		$this->expectApplied( 'string_hook_to_add' )->once()->andReturnString();
+
+		add_filter( 'string_hook_to_add', fn () => 'string' );
+
+		$this->assertIsString( apply_filters( 'string_hook_to_add', 'not_string' ) );
```
Comment (mogmarsh) [inline tests/testing/concerns/test-interacts-with-hooks.php:102]: $this->assertIsString( apply_filters( 'string_hook_to_add', null ) );

**Review by mogmarsh — COMMENTED** [review]
I actually found a few things here... overall 👍 but check out my comments.

File: src/mantle/database/model/registration/trait-register-meta.php [hunk 3]
```diff
@@ -54,9 +56,20 @@ public static function boot_register_meta() {
...
-		}
+		$args = array_merge(
+			event(
+				'mantle_register_meta_default_args',
+				[
+					[
+						'object_subtype' => static::get_object_name(),
+						'show_in_rest'   => true,
+						'single'         => true,
+						'type'           => 'string',
```
[the remark this answers is not in the document]
Comment (srtfisher) [reply src/mantle/database/model/registration/trait-register-meta.php:67]: $args can still override it, these are just the defaults

**Pull request merged**